providers/plunk/src/lib/plunk.provider.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IEmailOptions,
+  IEmailProvider,
+  ICheckIntegrationResponse,
+  IEmailEventBody,
+  CheckIntegrationResponseEnum,
+} from '@novu/stateless';
+
+import Plunk from '@plunk/node';
+import { IPlunkResponse } from './plunk.interface';
+
+export class PlunkEmailProvider implements IEmailProvider {
+  id = 'plunk';
+  channelType = ChannelTypeEnum.EMAIL as ChannelTypeEnum.EMAIL;
+
+  private plunk: Plunk;
+
+  constructor(
+    private config: {
+      apiKey: string;
+    }
+  ) {
+    this.plunk = new Plunk(this.config.apiKey);
+  }
+  getMessageId?: (body: any) => string[];
+  parseEventBody?: (body: any, identifier: string) => IEmailEventBody;
+  async checkIntegration(
+    options: IEmailOptions
+  ): Promise<ICheckIntegrationResponse> {
+    try {
+      const response: IPlunkResponse = await this.plunk.emails.send({
+        to: options.to,
+        subject: options.subject,
+        body: options.html || options.text,
+      });
+
+      return {
+        success: response.success,
+        message: 'Integrated successfully!',
+        code: CheckIntegrationResponseEnum.SUCCESS,
+      };
+    } catch (error) {
+      return {
+        success: false,
+        message: error?.message,
+        code: CheckIntegrationResponseEnum.FAILED,
+      };
+    }
+  }
+
+  async sendMessage(
+    options: IEmailOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const response: IPlunkResponse = await this.plunk.emails.send({
+      from: options.from,
+      to: options.to,
+      subject: options.subject,
+      body: options.html || options.text,
+    });
+
+    return {
+      id: response.contact,
```

Comment:
I cannot find any reference to response shape in the provider docs.

What is `response.contact`? Is this id returned by provider?

Reply:
Founder of Plunk here. First of all really happy to see the open-source power. I went ahead and updated the docs because return shape was indeed missing for this feature. Plunk by default returns an array of contact IDs (https://docs.useplunk.com/api-docs/transactional#returns) because it is possible to send the same email to up to 5 contacts at once (https://docs.useplunk.com/api-docs/transactional#sending-to-multiple-recipients). 

A Plunk contact is an email that is shared over automations, transactional emails and campaigns. We return unique identifiers so that our users can set up triggers, workflows etc in their own applications across different types of emails.

Reply:
@michaldziuba03 The provider has updated their docs. I have also updated my PR to return the ID